Common/Data/Custom/DailyFx.cs

```diff
@@ -0,0 +1,323 @@
+                }
+            }
+            catch 
+            { }
+
+            return dailyfx;
+        }
+
+
+        private static string RemoveSpecialCharacters(string str)
+        {
+            return Regex.Replace(str, "[^a-zA-Z0-9_.]+", "", RegexOptions.Compiled);
+        }
+
+        /// <summary>
+        /// Get the date search string for the quarter.
+        /// </summary>
+        /// <param name="date"></param>
+        /// <returns></returns>
+        private string GetQuarter(DateTime date)
+        {
+            var start = date.ToString("yyyy", CultureInfo.InvariantCulture);
+            var end = start;
+
+            if (date.Month < 4)
+            {
+                start += "0101";
+                end += "03312359"; 
+            }
+            else if (date.Month < 6)
+            {
+                start += "0401";
+                end += "06302359";
+            }
+            else if (date.Month < 9)
+            {
+                start += "0701";
+                end += "09302359";
+            }
+            else
+            {
+                start += "1001";
+                end += "12312359";
+            }
+            return "&startdate=20160526&format=json_pretty&enddate=201605262359"; // string.Format("&startdate={0}&enddate={1}", start, end);
+        }
+
+        /// <summary>
+        /// Pretty format output string for the DailyFx.
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            return string.Format("DailyFx [{0} {1} {2} {3} {4}]", Time.ToString("u"), Title, Currency, Importance, Meaning);
+        }
+    }
+
+    /// <summary>
+    /// FXDaily Importance Assignment.
+    /// </summary>
+    public enum FxDailyImportance
+    {
+        /// <summary>
+        /// Low importance
+        /// </summary>
+        [JsonProperty(PropertyName = "low")]
+        Low,
+
+        /// <summary>
+        /// Medium importance
+        /// </summary>
+        [JsonProperty(PropertyName = "medium")]
+        Medium,
+
+        /// <summary>
+        /// High importance
+        /// </summary>
+        [JsonProperty(PropertyName = "high")]
+        High
+    }
+
+    /// <summary>
+    /// What is the meaning of the event?
+    /// </summary>
+    public enum FxDailyMeaning
+    {
+        /// <summary>
+        /// The impact is perceived to be neutral.
+        /// </summary>
+        [JsonProperty(PropertyName = "NONE")]
+        None,
+
+        /// <summary>
+        /// The economic impact is perceived to be better.
+        /// </summary>
+        [JsonProperty(PropertyName = "TRUE")]
+        Better,
+
+        /// <summary>
+        /// The economic impact is perceived to be worse.
+        /// </summary>
+        [JsonProperty(PropertyName = "FALSE")]
+        Worse
+    }
+
+    /// <summary>
+    /// Helper to parse the Daily Fx API.
+    /// </summary>
+    public class DailyFxMeaningEnumConverter : JsonConverter
+    {
```

Comment:
Yeah strangely? Kept throwing errors and silently eating packets from the JSON. With the converter worked fine though. The test date 26th May had 40 events / GMT timing with one not specifying a time (https://www.dailyfx.com/calendar)